src/foam/nanos/ruler/predicate/IsClassPredicate.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler.predicate',
+  name: 'IsClassPredicate',
+
+  documentation: 'Returns true if new object is class of',
+
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+
+  javaImports: [
+    'static foam.mlang.MLang.*'
+  ],
+  properties: [
+    {
+      class: 'Class',
+      name: 'of',
+      documentation: 'class that we want the object to be an class of'
+    }
+  ],
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        return EQ(
```

Comment:
This is going to recreate all of the objects forming the mlang for each predicate check, which could be a lot.
Better to store this mlang in a property (or extend ProxyPredicate and use delegate).  This will generate a lot fewer objects and be much faster.